src/hotspot/cpu/x86/x86.ad

```diff
@@ -10007,7 +10007,7 @@ instruct mask_opers_evex(kReg dst, kReg src1, kReg src2, kReg kscratch) %{
   ins_encode %{
     const MachNode* mask1 = static_cast<const MachNode*>(this->in(this->operand_index($src1)));
     const MachNode* mask2 = static_cast<const MachNode*>(this->in(this->operand_index($src2)));
-    assert(0 == Type::cmp(mask1->bottom_type(), mask2->bottom_type()), "");
+    assert(Type::cmp(mask1->bottom_type(), mask2->bottom_type()), "");
```

Comment:
While at it, you could add a message like "should be false" for good practice.

Reply:
Well, I would say it should be "types must be equal".

Reply:
This is a good idea, I've made this change.